src/metatensor/models/experimental/alchemical_model/train.py

```diff
@@ -102,13 +102,13 @@ def train(
     model.set_normalization_factor(average_number_of_atoms)
     model.set_basis_normalization_factor(average_number_of_neighbors)
 
-    logger.info(f"Training on device {device_str}")
-    if device_str == "gpu":
-        device_str = "cuda"
-    device = torch.device(device_str)
-    if device.type == "cuda":
-        if not torch.cuda.is_available():
-            raise ValueError("CUDA is not available on this machine.")
+    if len(devices) > 1:
```

Comment:
Shouldn't this be all go into the `utils.device` functions? I would not like to have any checks in the `train.py` anymore.

rather we add more metadata in the head of the architecture which we then use outside.

Reply:
Yes, you're right. Not these tests. I'll remove them